### PR TITLE
Prac 05 code review request

### PR DESCRIPTION
**Pull request by al-c7**
Hi @danbrooks784, could you please review my code for Practical 5. For, `wimbledon.py`  in particular could you check that the functions used are correct, i.e. they obey SRP and are the 4 functions Lindsay intended in the Practical description. Thanks :)

### CODE REVIEW

File: prac_05/state_names.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+"""
+CP1404/CP5632 Practical
+State names in a dictionary
+File needs reformatting
+"""
+
+# TODO: Reformat this file so the dictionary code follows PEP 8 convention
```
Comment (danbrooks784) [inline prac_05/state_names.py]: Left a TODO comment here

File: prac_05/emails.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    email_to_name = {}
+    email = input("Email: ")
+    while email != "":
+        name = extract_name(email)
+        yes_no_input = input(f"Is your name {name}? (Y/n): ")
+        if yes_no_input == 'Y' or yes_no_input == '':
+            email_to_name[email] = name
+        else:
+            name = input("Name: ")
+            email_to_name[email] = name
```
Comment (danbrooks784) [inline prac_05/emails.py]: Could simplify this decision since they both end in the name being added

File: prac_05/emails.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        email = input("Email: ")
+    for email, name in email_to_name.items():
+        print(f"{name} ({email})")
+
+
+def extract_name(email):
+    name = email.split('@')[0]
+    names = name.split(".")
+    names = [name.title() for name in names]
+    name = " ".join(names)
```
Comment (danbrooks784) [inline prac_05/emails.py]: Could probably simplify this function as well

**Review by danbrooks784 — APPROVED** [review]
Your functions in `wimbledon.py` look good! They're different from the way I approached the practical but they look like they obey SRP and work according to the example output. There were a couple of small things you could simplify in `emails.py` though. Nice work overall!

File: prac_05/state_names.py (hunk 1 above)
Comment (al-c7) [reply prac_05/state_names.py]: Fixed.

File: prac_05/emails.py (hunk 2 above)
Comment (al-c7) [reply prac_05/emails.py]: Good find, refactored to make it one line.

File: prac_05/emails.py (hunk 3 above)
Comment (al-c7) [reply prac_05/emails.py]: Thanks, refactored function to two lines now.

**Pull request merged**